instaloader/instaloader.py

```diff
@@ -580,45 +574,41 @@ def _all_already_downloaded(path_base, is_videos_enumerated) -> bool:
                         if '{filename}' in self.filename_pattern:
-                            suffix = None
-                        if self.download_pictures and (not sidecar_node.is_video or self.download_video_thumbnails):
-                            # pylint:disable=cell-var-from-loop
-                            filename = self.__prepare_filename(filename_template, lambda: sidecar_node.display_url)
-                            # Download sidecar picture or video thumbnail (--no-pictures implies --no-video-thumbnails)
-                            downloaded &= self.download_pic(filename=filename, url=sidecar_node.display_url,
-                                                            mtime=post.date_local, filename_suffix=suffix)
-                        if sidecar_node.is_video and self.download_videos:
-                            # pylint:disable=cell-var-from-loop
-                            filename = self.__prepare_filename(filename_template, lambda: sidecar_node.video_url)
-                            # Download sidecar video if desired
-                            downloaded &= self.download_pic(filename=filename, url=sidecar_node.video_url,
-                                                            mtime=post.date_local, filename_suffix=suffix)
-                else:
-                    downloaded = False
+                            suffix = ''
+                        # pylint:disable=cell-var-from-loop
+                        filename2 = self.__prepare_filename(filename_template, lambda: sidecar_node.display_url)
+                        # Download sidecar picture or video thumbnail (--no-pictures implies --no-video-thumbnails)
+                        downloaded &= self.download_pic(filename=filename2, url=sidecar_node.display_url,
+                                                        mtime=post.date_local, filename_suffix=suffix)
+                    if sidecar_node.is_video and self.download_videos:
+                        suffix = str(edge_number)
+                        if '{filename}' in self.filename_pattern:
+                            suffix = ''
+                        # pylint:disable=cell-var-from-loop
+                        filename2 = self.__prepare_filename(filename_template, lambda: sidecar_node.video_url)
+                        # Download sidecar video if desired
+                        downloaded &= self.download_pic(filename=filename2, url=sidecar_node.video_url,
+                                                        mtime=post.date_local, filename_suffix=suffix)
         elif post.typename == 'GraphImage':
             # Download picture
             if self.download_pictures:
-                downloaded = (not _already_downloaded(filename + ".jpg") and
-                              self.download_pic(filename=filename, url=post.url, mtime=post.date_local))
+                downloaded = self.download_pic(filename=filename, url=post.url, mtime=post.date_local)
```

Comment:
This not _already_downloaded check was to make sure that the URL is not accessed (which can trigger an extra request) if the file is already downloaded.

Reply:
Ahh, now I see it. The problem was for logged in users, because of a query needed for `_iphone_struct` (high-res images). As I told before, I run it with default parameters, so not logged in. That's why that query didn't catch my eye.

Inside the `Post` constructor there has already been code to reload the `iphone_struct`. But isn't the real problem that no one ever loads the structure from disk!? I'm doing it now, so `iphone_struct` should be loaded too, isn't it? Can you please grab the branch and verify it?

Reply:
> I'm doing it now, so `iphone_struct` should be loaded too, isn't it?

`iphone_struct` is saved and loaded too with `save_structure_to_file()`/`load_structure_from_file()`.